### PR TITLE
optimize agg queries

### PR DESCRIPTION
**Pull request by OriHoch**
this should reduce the query times without changing the results

### CODE REVIEW

**Comment by ShayAdler** [comment]
@OriHoch why was only a day of tolerance problematic?

**Comment by OriHoch** [comment]
the gtfs date is the date that the data was collected on, it's not necesarily related to the date of the actual ride

I noticed it on the site where it showed less data with the optimization then without..

**Comment by ShayAdler** [comment]
Alright I see it in the code now, but I was sure that this is the day we downloaded it from the MOT FTP, the therefore we still have reference to the original date. Can you help me here with understanding the gtfs loading process? (when are we downloading from the ftp? when is the original date is getting lost in the way?)

**Comment by ShayAdler** [comment]
We actually [rely on that](https://github.com/hasadna/open-bus-stride-etl/blob/24ab0b110650b05e8fa272d93dbda8c9615b9d7f/open_bus_stride_etl/siri/update_rides_gtfs.py#L39) when setting the `siri_ride.scheduled_time_gtfs_ride_id` to optimize the search there as well, maybe we shouldn't?
Maybe best would be adding an ETL that will update the route date based on the most common gtfs_rides.start_time? I definitely feel like I understand nothing again 😅 

Basically AFAIK, we only use that date to narrow down searches, as the date that really matters is the ride's.

I think it will help me to look at the raw data, do we store it like we store siri?

**Comment by OriHoch** [comment]
yes, the raw data is availab,e see here - https://github.com/hasadna/open-bus-pipelines/blob/main/STRIDE_ETL_PROCESSES.md#gtfs-etl-download-upload

**Comment by ShayAdler** [comment]
@OriHoch can you expand on how did you see this bug?
I wan unable to find such cases in the data (this returns 0 results) - 

``` sql
-- get all rides in which the diff between the gtfs_route.date and the gtfs_ride.start_time is more than 1 day. Sometimes we do have 1 day diff, mostly when the ride hour is late.
select EXtract(epoch from date_trunc('day', gtfs_ride.start_time) - gr.date) , * from gtfs_ride join gtfs_route gr on gtfs_ride.gtfs_route_id = gr.id
where date_trunc('day', gtfs_ride.start_time) != gr.date
  and EXtract(epoch from date_trunc('day', gtfs_ride.start_time) - gr.date) not in (-86400, 86400)
and date_trunc('day', gtfs_ride.start_time) between '2023-10-01' and '2023-10-30'
```

**Comment by OriHoch** [comment]
This is the original complaint - https://hasadna.slack.com/archives/C1M8YT5AN/p1701675411278589
this change fixed it
